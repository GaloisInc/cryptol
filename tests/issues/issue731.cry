type constraint T n = (fin n, n >= 1)

type constraint Both p q = (p, q)

type constraint Fin2 i j = Both (fin i) (fin j)

type constraint i <=> j = (i <= j, i >= j)
