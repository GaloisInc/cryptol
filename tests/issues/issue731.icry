:l issue731.cry
:browse issue731
