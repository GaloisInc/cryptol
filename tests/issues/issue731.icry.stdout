Loading module Cryptol
Loading module Cryptol
Loading module Main
Constraint Synonyms
===================

  Public
  ------

    type constraint (<=>) i j = (i <= j, i >= j)
    type constraint Both p q = (p, q)
    type constraint Fin2 i j = Both (fin i) (fin j)
    type constraint T n = (fin n, n >= 1)

